File: BE/src/main/java/org/social/model/SocketEvent.java
package org.social.model;

import lombok.Data;

@Data
public class SocketEvent {
    private Event event;
    private Object data;

    public SocketEvent() {
    }

    public SocketEvent(Event event) {
        this.event = event;
    }

    public enum Event {
        CONNECTED,
        JOIN_ROOM,
        ERROR,
        FETCH_MESSAGE,
        DISCONNECTED,
        SEND_MESSAGE,
        DELETE_MESSAGE
    }
}

File: Mobile/Flutter/qcsocial/lib/model/SocketEvent.dart
import 'package:json_annotation/json_annotation.dart';

part 'SocketEvent.g.dart';

@JsonSerializable()
class SocketData {
  @JsonKey(required: true)
  SocketEvent? event;
  dynamic data;

  SocketData(this.event, {this.data});

  factory SocketData.fromJson(Map<String, dynamic> json) => _$SocketDataFromJson(json);

  Map<String, dynamic> toJson() => _$SocketDataToJson(this);
}

enum SocketEvent {
  CONNECTED,
  JOIN_ROOM,
  ERROR,
  FETCH_MESSAGE,
  DISCONNECTED,
  SEND_MESSAGE,
  DELETE_MESSAGE
}
